src/core_plugins/elasticsearch/lib/__tests__/check_es_version.js

```
@@ -54,40 +56,46 @@ describe('plugins/elasticsearch', function () {
 
       const client = server.plugins.elasticsearch.client;
       client.nodes.info = sinon.stub().returns(Promise.resolve({ nodes: nodes }));
-
     }
 
-    it('passes with single a node that matches', function () {
-      setNodes('1.4.3');
-      return checkEsVersion(server);
+    it('returns true with single a node that matches', done => {
+      setNodes('5.1.0');
+      checkEsVersion(server, KIBANA_VERSION).then(result => {
+        expect(result).to.be(true);
+        done();
+      });
     });
 
-    it('passes with multiple nodes that satisfy', function () {
-      setNodes('1.4.3', '1.4.4', '1.4.3-Beta1');
-      return checkEsVersion(server);
+    it('returns true with multiple nodes that satisfy', done => {
+      setNodes('5.1.0', '5.2.0', '5.1.1-Beta1');
+      checkEsVersion(server, KIBANA_VERSION).then(result => {
+        expect(result).to.be(true);
+        done();
+      });
     });
 
-    it('fails with a single node that is out of date', function () {
-      setNodes('1.4.4', '1.4.2', '1.4.5');
-
-      checkEsVersion(server)
-      .catch(function (e) {
+    it('throws an error with a single node that is out of date', done => {
+      // 5.0.0 ES is too old to work with a 5.1.0 version of Kibana.
+      setNodes('5.1.0', '5.2.0', '5.0.0');
+      checkEsVersion(server, KIBANA_VERSION)
+      .catch(e => {
         expect(e).to.be.a(SetupError);
+        done();
       });
     });
 
-    it('fails if that single node is a client node', function () {
+    it('fails if that single node is a client node', done => {
       setNodes(
-        '1.4.4',
-        { version: '1.4.2', attributes: { client: 'true' } },
-        '1.4.5'
+        '5.1.0',
+        '5.2.0',
+        { version: '5.0.0', attributes: { client: 'true' } },
       );
 
-      checkEsVersion(server)
-      .catch(function (e) {
+      checkEsVersion(server, KIBANA_VERSION)
+      .catch(e => {
         expect(e).to.be.a(SetupError);
+        done();
```

The errors are not being propagated here